intake/catalog/base.py

```diff
@@ -298,26 +302,36 @@ def __init__(self, url, http_args={}, page_size=None, **kwargs):
         page_size : int, optional
             The number of entries fetched at a time during iteration.
             Default is None (no pagination; fetch all entries in bulk).
+        query : Iterable or None
+            List from a sequence of progressive searches. Each element is
+            expected to be a valid argument to Catalog.search() for this
```

Comment:
In order to accommodate multiple args/kwargs (rather than just one arg, `query`, in my initial PR) `query` now expects an even _more_ persnickety structure. If you prefer, we could remove the `query` arg from `__init__` and add a new classmethod, `RemoteCatalog.from_query`, to be called by `RemoteCatalog.serach()` and perhaps by sophisticated users.